dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/Util.java

```diff
@@ -276,6 +285,23 @@ static final boolean isWrapperClassFor(Class<?> primitive, Class<?> cl) {
         NON_ENTITY_RESULT_TYPES.add(String.class);
     }
 
+    /**
+     * List of names of repository method life cycle annotations.
+     *
+     * @param producer producer of the repository bean, from which it can be
+     *                     determined if the repository is stateful or stateless.
+     */
+    @Trivial
+    static String lifeCycleAnnoNames(RepositoryProducer<?> producer) {
+        Set<Class<? extends Annotation>> annoClasses = producer.provider().compat //
+                        .lifeCycleAnnoTypes(producer.stateful());
+
+        StringBuilder b = new StringBuilder();
+        for (Class<?> annoClass : annoClasses)
+            b.append(b.isEmpty() ? "" : ", ").append(annoClass.getSimpleName());
+        return b.toString();
+    }
```

Comment:
```suggestion
    /**
     * List of names of repository method life cycle annotations.
     * Enclosed in brackets and delimited by comma.
     *
     * @param producer producer of the repository bean, from which it can be
     *                     determined if the repository is stateful or stateless.
     */
    @Trivial
    static String lifeCycleAnnoNames(RepositoryProducer<?> producer) {
        Set<Class<? extends Annotation>> annoClasses = producer.provider().compat //
                        .lifeCycleAnnoTypes(producer.stateful());

        return annoClasses.stream()
                   .map(c -> c.getSimpleName)
                   .collect(Collectors.joining(", ", "[", "]"));
    }
```

nitpick: This will show up in trace as a non-list formatted string which can result in incorrect grammar. i.e. `... only annotations Delete, Update, Insert, Save allowed.`